res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,921 @@
+        }
+    });
+
+    // Performance Pads
+    this.padGrid = new Prime4.PadSection(this, midiChannel - 4);
+
+    // Tempo Fader
+    this.tempoFader = new components.Pot({
+        midi: [0xB0 + midiChannel, 0x1F],
+        inKey: "rate",
+        invert: true,
+    });
+
+    // LED indicator when pitch fader is at centre.
+    this.tempoFeedback = new components.Button({
+        midi: [0x90 + midiChannel, 0x34],
+        outKey: "rate",
+        on: 0x02,
+        off: 0x00,
+        outValueScale: function(value) {
+            return value === 0 ? this.on : this.off;
+        },
+    });
+
+    // Keylock Button
+    this.keylockButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x22],
+        key: "keylock",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    this.vinylButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x23],
+        type: components.Button.prototype.types.toggle,
+        input: function(channel, control, value, status, _group) {
+            if (!this.isPress(channel, control, value, status)) {
+                return;
+            }
+            theDeck.jogWheel.vinylMode = !theDeck.jogWheel.vinylMode;
+            this.output(theDeck.jogWheel.vinylMode);
+        },
+        trigger: function() {
+            this.output(theDeck.jogWheel.vinylMode);
+        },
+    });
+
+    // Jog Wheel
+    this.jogWheel = new components.JogWheelBasic({
+        deck: script.deckFromGroup(this.currentDeck),
+        wheelResolution: 1000,
+        alpha: 1/8,
+        beta: 1/8/32,
+        rpm: 33 + 1/3,
+        // Instead of relative movements between this and the last position,
+        // the controller reports the absolute position of the wheel with
+        // 14-bit precision. Because of that, we need to reconstruct the value
+        // and then transform it into the relative directions expected by Mixxx.
+        inputWheelMSB: function(_channel, _control, value, _status, _group) {
+            this.wheelMSB = value;
+        },
+        inputWheelLSB: function(channel, control, value, status, group) {
+            this.inputWheel(channel, control, (this.wheelMSB << 7) + value, status, group);
+        },
+        previousPosition: null,
+        wrappingValue: Math.pow(2, 14),
+        relativeFromAbsolute: function(value) {
+            // The first value of the controller will probably be random
+            // and thus we just have to swallow it until we have the second value
+            // to find the difference
+            if (this.previousPosition === null) {
+                this.previousPosition = value;
+                return 0;
+            }
+            // This finds the shortest distance between the current value
+            // and the last one, and preserves the orientation
+            const delta = value - this.previousPosition;
+            let remainder = ((delta % this.wrappingValue) + this.wrappingValue) % this.wrappingValue;
```

Comment:
```suggestion
            let remainder = script.posMod(delta, this.wrappingValue);
```

Reply:
```
Uncaught exception: file:///home/whanake/.mixxx/controllers/Denon-Prime-4-scripts.js:589: TypeError: Property 'posMod' of object function script() { [native code] } is not a function
Backtrace: relativeFromAbsolute@file:///home/whanake/.mixxx/controllers/Denon-Prime-4-scripts.js:589
inputWheel@file:///home/whanake/.mixxx/controllers/Denon-Prime-4-scripts.js:601
inputWheelLSB@file:///home/whanake/.mixxx/controllers/Denon-Prime-4-scripts.js:573
@:1
```

Reply:
It might be that your local branch is based on an out-of-date state. Can you `git merge upstream/main`?